langchain/spark_sql.py

```diff
@@ -0,0 +1,111 @@
+from typing import Any, Iterable, List, Optional
+
+from pyspark.sql import DataFrame, Row, SparkSession
+
+from langchain import SQLDatabase
+
+
+class SparkSQL(SQLDatabase):
+    def __init__(
+        self,
+        spark_session: Optional[SparkSession] = None,
+        catalog: Optional[str] = None,
+        schema: Optional[str] = None,
+        ignore_tables: Optional[List[str]] = None,
+        include_tables: Optional[List[str]] = None,
+        sample_rows_in_table_info: int = 3,
+    ):
+        self._spark = (
+            spark_session if spark_session else SparkSession.builder.getOrCreate()
+        )
+        if catalog is not None:
+            self._spark.catalog.setCurrentCatalog(catalog)
+        if schema is not None:
+            self._spark.catalog.setCurrentDatabase(schema)
+
+        self._all_tables = set(self._get_all_table_names())
+        super()._init_table_options(
+            ignore_tables=ignore_tables,
+            include_tables=include_tables,
+            sample_rows_in_table_info=sample_rows_in_table_info,
+        )
+
+    @classmethod
+    def from_uri(
+        cls, database_uri: str, engine_args: Optional[dict] = None, **kwargs: Any
+    ) -> SQLDatabase:
```

Comment:
can type as SparkSQL. just add this at top
```python
from __future__ import annotations
```

Reply:
Thanks, updated